### PR TITLE
dot-http: init at 0.2.0

### PR DESCRIPTION
**Pull request by mredaelli**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/dot-http/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, rustPlatform, fetchFromGitHub, openssl, pkg-config, lib, libiconv, darwin }:
```
Comment (danieldk) [inline pkgs/development/tools/dot-http/default.nix:1]:
```suggestion
{ stdenv, rustPlatform, fetchFromGitHub, pkg-config, openssl, libiconv, Security }:
```

`lib` is not needed (see below), it is somewhat of a convention to order native build inputs before build inputs.

File: pkgs/development/tools/dot-http/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    owner = "bayne";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1s2q4kdldhb5gd14g2h6vzrbjgbbbs9zp2dgmna0rhk1h4qv0mml";
+  };
+
+  cargoSha256 = "0an3hskq1k2j4gdn8wvhfb9pqsc34ibs5bv7sjznkp5jma6fdr9w";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [
```
Comment (danieldk) [inline pkgs/development/tools/dot-http/default.nix:17]:
```suggestion

  buildInputs = [ openssl ] ++ stdenv.lib.optionals stdenv.isDarwin [
```

File: pkgs/development/tools/dot-http/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    libiconv darwin.apple_sdk.frameworks.Security
```
Comment (danieldk) [inline pkgs/development/tools/dot-http/default.nix:18]:
```suggestion
    libiconv Security
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -985,6 +985,8 @@ in
 
   dpt-rp1-py = callPackage ../tools/misc/dpt-rp1-py { };
 
+  dot-http = callPackage ../development/tools/dot-http { };
```
Comment (danieldk) [inline pkgs/top-level/all-packages.nix:988]:
```suggestion
  dot-http = callPackage ../development/tools/dot-http {
    inherit (darwin.apple_sdk.frameworks) Security;
  };
```

**Review by danieldk — CHANGES_REQUESTED** [review]
Looks good, some small changes to make the derivation more canonical.

**Comment by mredaelli** [comment]
@danieldk thank you for the review!

Can I use this (new?) "commit suggestion" feature and then you squash the commits, of is it better if I mirror them locally and rebase? 

Or any other workflow that's easiest for you (I mean, if you can and want commit them yourself from here, I'm more than happy).

**Comment by danieldk** [comment]
> Can I use this (new?) "commit suggestion" feature and then you squash the commits, of is it better if I mirror them locally and rebase?

It's best to mirror them locally, but you should be able to get best from both worlds. First use the commit suggestions in the GitHub interface, then something like:

```
$ git pull --rebase
$ git rebase -i master
# Mark everything but the first commit squash, save and quit
$ git log # Verify that the history looks ok
$ git push -f
```

**Comment by mredaelli** [comment]
Will do, thanks. Just wanted to make sure I'm not going to drive CI jobs crazy :)

**Review by danieldk — APPROVED** [review]
LGTM, thanks a lot!

Result of `nixpkgs-review pr 96537` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- dot-http
</details>

**Pull request merged**